sstables_loader.cc

```diff
@@ -168,48 +179,30 @@ class tablet_sstable_streamer : public sstable_streamer {
 };
 
 inet_address_vector_replica_set sstable_streamer::get_endpoints(const dht::token& token, bool primary_replica_only) const {
-    auto current_targets = _erm->get_natural_endpoints(token);
-    if (primary_replica_only && current_targets.size() > 1) {
-        current_targets.resize(1);
+    if (primary_replica_only) {
+        return get_primary_endpoints(token);
     }
+    auto current_targets = _erm->get_natural_endpoints_without_node_being_replaced(token);
+    auto pending = _erm->get_pending_endpoints(token);
+    // Helps with intra-node migration, where we won't stream to same node twice.
```

Comment:
For intra-node migration there is no pending endpoint, so deduplication is not necessary.

Reply:
I was also mimicking what we have in storage proxy:

```
    // filter out natural_endpoints from pending_endpoints if the latter is not yet updated during node join
    // should never happen, but better to be safe
    auto itend = boost::range::remove_if(pending_endpoints, [&natural_endpoints] (gms::inet_address& p) {
        return boost::range::find(natural_endpoints, p) != natural_endpoints.end();
    });
```

do you think it's better to remove this deduplication code? and instead just insert pending endpoints unconditionally?

Reply:
but the actual reason I did it was thinking you would insert pending replica in same node as pending, but given that's not the case, I'll remove this. thanks